packages/vscode-extension/src/debugging/DebugAdapter.ts

```diff
@@ -144,6 +141,12 @@ export class DebugAdapter extends DebugSession {
         case "Debugger.scriptParsed":
           const sourceMapURL = message.params.sourceMapURL;
 
+          if (message.params.url) {
```

Comment:
this deserves some explanation. what does this check do, how often do we execute onDebuggerReady